api/server/router/system/system_routes.go

```diff
@@ -89,25 +89,60 @@ func (s *systemRouter) getVersion(ctx context.Context, w http.ResponseWriter, r
 	return httputils.WriteJSON(w, http.StatusOK, info)
 }
 
+// DiskUsageObject represents an object type used for disk usage query filtering.
+type DiskUsageObject string
+
+const (
+	// ContainerObject represents a container DiskUsageObject.
+	ContainerObject DiskUsageObject = "container"
+	// ImageObject represents an image DiskUsageObject.
+	ImageObject DiskUsageObject = "image"
+	// VolumeObject represents a volume DiskUsageObject.
+	VolumeObject DiskUsageObject = "volume"
+	// BuildCacheObject represents a build-cache DiskUsageObject.
+	BuildCacheObject DiskUsageObject = "build-cache"
+)
+
 func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	var objectTypes map[DiskUsageObject]struct{}
+	if ss, ok := r.Form["types"]; ok {
+		objectTypes = make(map[DiskUsageObject]struct{}, 4)
+		for _, s := range ss {
```

Comment:
nit: the `s` variable collides with the `s *systemRouter` receiver, so perhaps we should pick a different name, e.g. 

```suggestion
		for _, ts := range ss {
```